### DISCUSSION TITLE
Next.js: Error: ENOENT: no such file or directory, lstat '/vercel/path0/node_modules/@swc/wasm/package.json'

### DISCUSSION THREAD

**Discussion by nbouvrette** [vercel/community#1209] [Help] [answered]
### Summary

We started getting this error when GitHub commits trigger Vercel Next.js deployments.

We are not using SWC for our project as we have Babel plugins and SWC is also not in our package.json.

Our build is failing with this error, but when we redeploy, it works.. any suggestions?

### Additional information

This is from our Vercel console:

```sh
λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
--
14:32:21.490 | ○  (Static)  automatically rendered as static HTML (uses no initial props)
14:32:21.490 | ●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)
14:32:21.490 |  
14:32:21.643 | Error: ENOENT: no such file or directory, lstat '/vercel/path0/node_modules/@swc/wasm/package.json'
```

**Comment by amyegan** [comment, marked as answer]
Hi @nbouvrette! This sounds like a build cache problem. Depending on how your app is set up, it may be necessary to skip the build cache. You could add an env var called `VERCEL_FORCE_NO_BUILD_CACHE` with a value of `1` to skip the build cache for all new deployments. 

Do you have a public repo or [minimal reproduction](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that I can use to take a closer look? I would like to share it with the team so we can figure out the root cause and possibly include a solution link in the error message for this in the future 🙏

**Reply by mAAdhaTTah** [reply]
I experienced this problem on [this PR](https://github.com/mAAdhaTTah/frontend/pull/812) during [this build](https://vercel.com/maadhattah/frontend/QTL3sBisT4MEnZAjcRCUZLevEkCg). I just reran it without a build cache, which succeeded, so that does seem to be related.

**Comment by shaunchander** [comment]
@nbouvrette we're getting the same issue with our NextJS site as well, are you by chance using Nx to build the website?

For context, we are using Next 13 with SWC. We're also using `@nrwl/next` to build the website out. It seems like redeploys fix the site though we're looking for the root cause.

**Comment by nbouvrette** [comment]
@shaunchander sorry for the late reply - we are not using SWC since we have Babel plugin which is why this error is a bit odd for us.

@amyegan - @RMBraun on our end was looking into the issue - I think your cache setting mitigated the issue but I will let him provide an update.

**Reply by RMBraun** [reply]
@amyegan  After putting in the environmental variable to disable build caching, the builds have been passing first try. Thanks for the advice!

**Reply by nbouvrette** [reply]
@amyegan it still looks like there is some sort of bug somewhere.

I'm not sure what is the impact of not using the cache but it might be worthwhile investigating on how to fix the root cause on the Vercel side.

Thanks for your help again.

**Comment by natario1** [comment]
Got a similar issue today, can't find `@next/font`. Redeploying without the build cache succeeds. The new thing WRT previous deployment is that we are using turborepo now, in case it helps debugging.

**Comment by 26A0D355-AA5F-4621-8975-9184E8DF8C23** [comment]
Similar issue but redeploying without the existing build cache does not resolve the issue.

`ENOENT: ENOENT: no such file or directory, lstat '/vercel/path0/node_modules/sodium-native/tmp/lib/libsodium.so.23.2.0'`

**Reply by emalindahk** [reply]
I'm having the same exact issue.

**Reply by kumarabhirup** [reply]
+1

**Reply by craigboldy** [reply]
Did you find a fix for this? I'm getting the same error

**Comment by rashajkbx** [comment]
Hey yall, I was experiencing this error after removing old packages and adding new ones. I was able to clear this error by deleting my node modules and package lock and then reinstalling my dependencies. My next build went through correctly

**Reply by rashajkbx** [reply]
If you are using @swc/core make sure it is up to date because it will generate new packages on build that are newer than your version.

**Comment by carlosvegabuydepa** [comment]
We also experienced this out of the blue. An alternative to setting `VERCEL_FORCE_NO_BUILD_CACHE ` is to add the `--force ` option to the vercel deploy command, ie `vercel --force` 


https://vercel.com/docs/cli/deploy#force

**Reply by swarnava** [reply]
Hi,

Can you create a new thread so we can take a closer look? Thanks!

**Comment by rodsfreitas** [comment]
Having the same, with a very basic nextjs 13.4 project.

**Reply by amyegan** [reply]
Please [start a new discussion](https://github.com/vercel/community/discussions/new) and share your public repo or a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so we can take a closer look at this problem. 🙏
